lib/rspec/hive/matchers.rb

```diff
@@ -0,0 +1,29 @@
+require 'rspec/matchers'
+
+RSpec::Matchers.define :match_result_set do |expected|
+  match do |actual|
+    return false if expected.size != actual.size
+
+    expected.each.with_index.all? do |expected_row, i|
+      if expected_row.respond_to?(:each_pair)
```

Comment:
does it guarantee that expected_row.keys and expected_row.values will be returned in the same order?